src/lib/components/custom/theme-selector.svelte

```diff
@@ -0,0 +1,126 @@
+<script lang="ts">
+	import Input from '../ui/input.svelte'
+	import Radio from '../ui/radio.svelte'
+	import { ColorPalette } from 'carbon-icons-svelte'
+	import { withThemeStore } from '$lib/stores/theme.svelte'
+	import { calculateLuminance } from '@waku-objects/luminance'
+	import Typography from '../ui/typography.svelte'
+	import { getEffectiveColorMode } from '$lib/utils/colors'
+
+	let theme = withThemeStore()
+	let baseColor = $state(theme.baseColor)
+	let mode = $state(theme.mode)
```

Comment:
These should not be wrapped in `$state`. You can just use them directly and this way there is no need for the `$effect` on line 25.